src/_P103_Atlas_EZO_pH_ORP_EC.ino

```diff
@@ -0,0 +1,641 @@
+#include "_Plugin_Helper.h"
+
+#ifdef USES_P103
+
+// ########################################################################
+// ################## Plugin 103 : Atlas Scientific EZO pH ORP EC sensors #
+// ########################################################################
+
+// datasheet at https://atlas-scientific.com/files/pH_EZO_Datasheet.pdf
+// datasheet at https://atlas-scientific.com/files/ORP_EZO_Datasheet.pdf
+// datasheet at https://atlas-scientific.com/files/EC_EZO_Datasheet.pdf
+// works only in i2c mode
+
+# include "src/Helpers/Rules_calculate.h"
+
+# define PLUGIN_103
+# define PLUGIN_ID_103           103
+# define PLUGIN_NAME_103         "Environment - Atlas EZO pH ORP EC"
+# define PLUGIN_VALUENAME1_103   "SensorData"
+# define PLUGIN_VALUENAME2_103   "Voltage"
+# define UNKNOWN                 0
+# define PH                      1
+# define ORP                     2
+# define EC                      3
+
+# define FIXED_TEMP_VALUE        20   // Temperature correction for pH and EC sensor if no temperature is given from calculation
+
+boolean Plugin_103(byte function, struct EventStruct *event, String& string)
+{
+  boolean success = false;
+
+  byte board_type;
+  byte I2Cchoice;
+
+  switch (function)
+  {
+    case PLUGIN_DEVICE_ADD:
+    {
+      Device[++deviceCount].Number           = PLUGIN_ID_103;
+      Device[deviceCount].Type               = DEVICE_TYPE_I2C;
+      Device[deviceCount].VType              = Sensor_VType::SENSOR_TYPE_DUAL;
+      Device[deviceCount].Ports              = 0;
+      Device[deviceCount].PullUpOption       = false;
+      Device[deviceCount].InverseLogicOption = false;
+      Device[deviceCount].FormulaOption      = true;
+      Device[deviceCount].ValueCount         = 2;
+      Device[deviceCount].SendDataOption     = true;
+      Device[deviceCount].TimerOption        = true;
+      Device[deviceCount].GlobalSyncOption   = true;
+      break;
+    }
+
+    case PLUGIN_GET_DEVICENAME:
+    {
+      string = F(PLUGIN_NAME_103);
+      break;
+    }
+
+    case PLUGIN_GET_DEVICEVALUENAMES:
+    {
+      strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[0], PSTR(PLUGIN_VALUENAME1_103));
+      strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[1], PSTR(PLUGIN_VALUENAME2_103));
+      break;
+    }
+
+    case PLUGIN_WEBFORM_LOAD:
+    {
+      I2Cchoice  = PCONFIG(1);
+      addHtml(F("<TR><TD>Default I2C Address:</TD><TD>pH: 0x63, ORP: 0x62, EC: 0x64. The plugin is able to detect the type of device automatically.</TD></TR>"));
```

Comment:
Why not an `addFormNote` here?

Reply:
Yes that's good


> Just took a quick look through the code.
> Mainly markup hints to use existing functions to get the look and feel as uniform as possible throughout all plugins.
> And there are some serious bugs in there regarding missing zero termination of char arrays.

I will fix this during the weekend. What I did yesterday was the unification. The serious bugs you mentioned ar existing since 3 years. Perhaps this will explain that, sometimes, my ESP hangs. I will fix them, too.

Reply:
fixed